File: dsa/go/binarySearch.go
package main

import (
	"fmt"
)

func main() {
	a := []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
	ans := binarySearch(a, 11)
	fmt.Println(ans)
}

func binarySearch(a []int, key int) int {
	low, high := 0, len(a)-1

	for low <= high {
		mid := (low + high) / 2
		if a[mid] == key {
			return mid
		} else if a[mid] > key {
			high = mid - 1
		} else {
			low = mid + 1
		}
	}

	return -1
}

File: dsa/java/src/search/BinarySearch.java
package search;

import java.rmi.dgc.VMID;

public class BinarySearch {
    public static void main(String []args){
        int[] list = new int[10];
        list[0] = 1;
        list[1] = 2;
        list[2] = 3;
        list[3] = 4;
        list[4] = 5;
        list[5] = 6;
        list[6] = 7;
        list[7] = 8;
        list[8] = 9;
        list[9] = 10;

        int ans = binarySearch(list, 41);
        System.out.println(ans);
    }

    public static int binarySearch(int[] list, int key){
        int low = 0;
        int high = list.length - 1;

        while(low <= high){
            int mid = (int)Math.floor((high + low) / 2);
            if(list[mid] == key){
                return mid;
            }else if(list[mid] > key){
                high = mid - 1;
            }else{
                low = mid + 1;
            }
        }
        return -1;
    }
}
